awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -12,409 +13,562 @@
+    timeout = module.params.get('timeout')
+    use_fact_cache = module.params.get('use_fact_cache')
+    host_config_key = module.params.get('host_config_key')
+    ask_scm_branch_on_launch = module.params.get('ask_scm_branch_on_launch')
+    ask_diff_mode_on_launch = module.params.get('ask_diff_mode_on_launch')
+    ask_variables_on_launch = module.params.get('ask_variables_on_launch')
+    ask_limit_on_launch = module.params.get('ask_limit_on_launch')
+    ask_tags_on_launch = module.params.get('ask_tags_on_launch')
+    ask_skip_tags_on_launch = module.params.get('ask_skip_tags_on_launch')
+    ask_job_type_on_launch = module.params.get('ask_job_type_on_launch')
+    ask_verbosity_on_launch = module.params.get('ask_verbosity_on_launch')
+    ask_inventory_on_launch = module.params.get('ask_inventory_on_launch')
+    ask_credential_on_launch = module.params.get('ask_credential_on_launch')
+    survey_enabled = module.params.get('survey_enabled')
+    become_enabled = module.params.get('become_enabled')
+    diff_mode = module.params.get('diff_mode')
+    allow_simultaneous = module.params.get('allow_simultaneous')
+    custom_virtualenv = module.params.get('custom_virtualenv')
+    job_slice_count = module.params.get('job_slice_count')
+    webhook_service = module.params.get('webhook_service')
+    webhook_credential = module.params.get('webhook_credential')
+    survey_spec = module.params.get('survey_spec')
+    credentials = module.params.get('credentials')
+    labels = module.params.get('labels')
+    credential = module.params.get('credential')
+    vault_credential = module.params.get('vault_credential')
+    state = module.params.get('state')
+
+    # Deal with legacy credential and vault_credential
+    if vault_credential:
+        if credentials is None:
+            credentials = []
+        credentials.append(vault_credential)
+    if credential:
+        if credentials is None:
+            credentials = []
+        credentials.append(credential)
+
+    # Attempt to look up the related items the user specified (these will fail the module if not found)
+    inventory_id = None
+    if inventory is not None:
+        inventory_id = module.resolve_name_to_id('inventories', inventory)
+    project_id = None
+    if project is not None:
+        project_id = module.resolve_name_to_id('projects', project)
+    webhook_credential_id = None
+    if webhook_credential is not None:
+        webhook_credential_id = module.resolve_name_to_id('credentials', webhook_credential)
+    credentials_ids = None
+    if credentials is not None:
+        credentials_ids = []
+        for item in credentials:
+            credentials_ids.append(module.resolve_name_to_id('credentials', item))
+    labels_ids = None
+    if labels is not None:
+        labels_ids = []
+        for item in labels:
+            labels_ids.append(module.resolve_name_to_id('labels', item))
+
+    # Pull in additional posts
+    additional_posts = []
```

Comment:
Since they're both hitting some kind of related items, it's really mismatched that the additional posts is a list while the additional associations is a dict. It's also really hard to keep up with an implied nested schema.